RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.h

```diff
@@ -0,0 +1,199 @@
+    adc = std::min(adc, uint32_t(thePacking.max_adc));
+
+    return (row << thePacking.row_shift) | (col << thePacking.column_shift) | (adc << thePacking.adc_shift);
+  }
+
+  constexpr uint32_t pixelToChannel(int row, int col) {
+    constexpr pixelchannelidentifierimpl::Packing thePacking = packing();
+    return (row << thePacking.column_width) | col;
+  }
+
+}  // namespace pixelDetails
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  namespace pixelDetails {
+
+    class SiPixelRawToClusterKernel {
+    public:
+      class WordFedAppender {
+      public:
+        WordFedAppender();
+        ~WordFedAppender() = default;
+
+        WordFedAppender(uint32_t words);
+
+        void initializeWordFed(int fedId, unsigned int wordCounterGPU, const uint32_t* src, unsigned int length);
+
+        auto word() const { return word_; }
+        auto fedId() const { return fedId_; }
+
+      private:
+        cms::alpakatools::host_buffer<unsigned int[]> word_;
+        cms::alpakatools::host_buffer<unsigned char[]> fedId_;
+      };
+
+      SiPixelRawToClusterKernel() : nModules_Clusters_h{cms::alpakatools::make_host_buffer<uint32_t[], Platform>(3u)} {}
+
+      ~SiPixelRawToClusterKernel() = default;
+
+      SiPixelRawToClusterKernel(const SiPixelRawToClusterKernel&) = delete;
+      SiPixelRawToClusterKernel(SiPixelRawToClusterKernel&&) = delete;
+      SiPixelRawToClusterKernel& operator=(const SiPixelRawToClusterKernel&) = delete;
+      SiPixelRawToClusterKernel& operator=(SiPixelRawToClusterKernel&&) = delete;
+
+      void makeClustersAsync(bool isRun2,
+                             const SiPixelClusterThresholds clusterThresholds,
+                             const SiPixelMappingLayoutSoAConstView& cablingMap,
+                             const unsigned char* modToUnp,
+                             const SiPixelGainCalibrationForHLTSoAConstView& gains,
+                             const WordFedAppender& wordFed,
+                             SiPixelFormatterErrors&& errors,
+                             const uint32_t wordCounter,
+                             const uint32_t fedCounter,
+                             bool useQualityInfo,
+                             bool includeErrors,
+                             bool debug,
+                             Queue& queue);
+
+      void makePhase2ClustersAsync(const SiPixelClusterThresholds clusterThresholds,
+                                   SiPixelDigisLayoutSoAView& digis_view,
+                                   const uint32_t numDigis,
+                                   Queue& queue);
+
+      // std::pair<SiPixelDigisDevice, SiPixelClustersDevice> getResults() {
+      //   digis_d->setNModulesDigis(nModules_Clusters_h[0], nDigis);
+      //   clusters_d->setNClusters(nModules_Clusters_h[1], nModules_Clusters_h[2]);
+      //   return std::make_pair(std::move(*digis_d), std::move(*clusters_d));
+      // }
+
+      SiPixelDigisDevice&& getDigis() {
```

Comment:
The proper return type is just
```suggestion
      SiPixelDigisDevice getDigis() {
```
(same for `SiPixelClustersDevice` and `SiPixelDigiErrorsDevice` below)